lib/SPIRV/SPIRVInternal.h

```diff
@@ -1037,6 +1037,23 @@ bool lowerBuiltinVariableToCall(GlobalVariable *GV,
 // Transform all builtin variables into calls
 bool lowerBuiltinVariablesToCalls(Module *M);
 
+/// \brief Post-process OpenCL or SPIRV builtin function returning struct type.
+///
+/// Some builtin functions are translated to SPIR-V instructions with
```

Comment:
The comment is a bit confusing. Does it mean, that in SPV-IR we have a function, that have some structure return type and we replace it with a function of void type, or we already have a void function, which `postProcessBuiltinReturningStruct` function modifies to add `sret` attribute?

Reply:
Some SPIR-V instructions may have OpTypeStruct as the type of the result. SPIRVReader translates such instructions to functions that return struct via return value. No matter of chosen output IR format (SPV-IR or OCL), `postProcessBuiltinReturningStruct` makes a function returning struct via return value return it via first argument with sret attribute. It is the only way to return a struct from a function that SPIR-V translator accepts when reads input IR, I guess that is because clang emits functions returning struct this way.